### ISSUE TITLE
Create new tests project for dotnetcore

### ISSUE THREAD

**Issue by eByte23** [SapientGuardian/mysql-connector-net-netstandard#5]
Create a new project for dotnetcore, try to  utiliseold test as much as possible. 

Convert NUnit tests to XUnit so that they run under dotnet test with xunit runner.

**Comment by eByte23** [comment]
I'm doing this, just as a place holder to say we are doing :smile:

**Comment by SapientGuardian** [comment]
:+1:

**Comment by eByte23** [comment]
Should have something today i was trying to reutilse alot of the original test and multi target so i could it wouldn't break old tests, but i can't even build any of the previous projects in vs15.
And there is a bug in multiargeting with includes and excludes in project. json _sigh_

So ill just make it work rc2 ha in new tests

**Comment by SapientGuardian** [comment]
I spent a few minutes trying to make the existing tests work. It looks like there's a few problems, which shouldn't be insurmountable:
- Some tests use NUnit
- The XUnit tests target an old version of XUnit. Replacing IUseFixture with IClassFixture might be all that's needed here
- The tests were also built with ifdefs, particularly around RT, so we'd need to do the same treatment for our netstandard build (or netcoreapp10 in the case of tests, I'd imagine)

At this point, any tests would be good; so if writing your own is the quickest path to that, I'll take it. If Oracle ends up taking so long on their .net core build that a lot of people start to use this fork, it may be worth the effort to get the existing tests working.

**Comment by eByte23** [comment]
Yeah it's not to hard to reuse alot of the previous test I just could really test for sure that I would break them so I'm going to copy the to begin with.
You can reference nunitlite as a replacement for normal nunit

I was trying to future proof so it could be used in backwards compat. But i'm just going to make it work then look at that later.

**Comment by eByte23** [comment]
Just getting back to this test business....and tests are running looks like we have a lot failing haha....
BeginExecuteReader and EndExecuteReader don't work at all, the AsyncDelegate is not implemented in .net core throws an exception, I guess no one is using it haha
